qtpy/__init__.py

```diff
@@ -207,17 +211,26 @@ class PythonQtWarning(Warning):
 def _warn_old_minor_version(name, old_version, min_version):
     """Warn if using a Qt or binding version no longer supported by QtPy."""
     warning_message = (
-        "{name} version {old_version} is no longer supported upstream or "
+        "{name} version {old_version} is not supported upstream or "
         "by QtPy 2.0. To ensure your application works correctly with QtPy, "
         "please upgrade to {name} {min_version} or later.".format(
             name=name, old_version=old_version, min_version=min_version))
     warnings.warn(warning_message, PythonQtWarning)
 
 
-# Warn if using an End of Life, unsupported Qt API/binding
-if QT_VERSION and parse(QT_VERSION) < parse(QT_VERSION_MIN):
-    _warn_old_minor_version('Qt', QT_VERSION, QT_VERSION_MIN)
-if PYQT_VERSION and parse(PYQT_VERSION) < parse(PYQT_VERSION_MIN):
-    _warn_old_minor_version('PyQt', PYQT_VERSION, PYQT_VERSION_MIN)
-elif PYSIDE_VERSION and parse(PYSIDE_VERSION) < parse(PYSIDE_VERSION_MIN):
-    _warn_old_minor_version('PySide', PYSIDE_VERSION, PYSIDE_VERSION_MIN)
+# Warn if using an End of Life or unsupported Qt API/binding minor version
+if QT_VERSION:
+    if parse(QT_VERSION) < parse(QT5_VERSION_MIN):
+        _warn_old_minor_version('Qt5', QT_VERSION, QT5_VERSION_MIN)
+    elif parse('6') <= parse(QT_VERSION) < parse(QT6_VERSION_MIN):
```

Comment:
Checking @kumattau comments this vaidation looks a little bit confusing at first glance.

Maybe we could add some boolean constants like `QT5` and `QT6` (following the same spirit as the existing ones (`PYQT5`, `PYQT6`, `PYSIDE2` and `PYSIDE6`) where ony one is `True` at a time. I think that could help making the validation more clear, something like:

```python

elif QT6 and parse(QT_VERSION) < parse(QT6_VERSION_MIN):

```

 What do you think @CAM-Gerlach @kumattau ? Is adding this new constants a good compromise between clarity vs new constants/redundancy?

Reply:
I think @dalthviz 's idea is better from the perspective of consistency.

But since it is difficult to understand the different ranges represented by each constant,

* PYQT5 : True if PyQt5 is selected
* PYQT_VERSION : PYQT5 or PYQT6 version
* PYQT5_VERSION_MIN : minimum version of supported PyQt5
* PYQT5_VERSION is nothing.

(PYSIDE and QT are the same as PYQT)

If these constants are public API for users but PYQT5_VERSION and so on are not added,
I think it is better approach that the usage of related constants is explained clearly in the document like as follows:

```python
# check Qt6 version (need to check QT6 first)
if QT6 and parse(QT_VERSION) < parse(QT6_VERSION_MIN):
    print("cannot support or deprecated")

# check Qt5 version (need to check QT5 first)
if QT5 and parse(QT_VERSION) < parse(QT5_VERSION_MIN):
    print("cannot support  or deprecated")

# check PySide2 version (need to check PYSIDE2 first)
if PYSIDE2 and parse(PYSIDE_VERSION) < parse(PYSIDE2_VERSION_MIN)
    print("cannot support  or deprecated")

# check PySide6 version (need to check PYSIDE6 first)
if PYSIDE6 and parse(PYSIDE_VERSION) < parse(PYSIDE2_VERSION_MIN)
    print("cannot support  or deprecated")

# check PyQt5 version (need to check PYQT5 first)
if PYQT5 and parse(PYQT_VERSION) < parse(PYQT5_VERSION_MIN)
    print("cannot support  or deprecated")

# check PyQt6 version (need to check PYQT6 first)
if PYQT6 and parse(PYQT_VERSION) < parse(PYQT6_VERSION_MIN)
    print("cannot support  or deprecated")
```

Reply:
I agree with @dalthviz 's approach as well, would be useful to have those top-level constants and makes the code much cleaner while avoiding the extra complexity of the previous suggested approach. I also agree with explicitly checking the constants in each branch, for clarity and completeness, though the existence of the version constant has to be checked as well at least with the current `__init__` structure, as that is imported from the Qt bindings as well and may not be defined if all of the imports fail and it falls back to the default, causing a hard crash. I also separated the Qt and bindings checks by a line, to make the structure more clear. 

Here's what the final result looks like, which I've pushed here:

```python
# Warn if using an End of Life or unsupported Qt API/binding minor version
if QT_VERSION:
    if QT5 and (parse(QT_VERSION) < parse(QT5_VERSION_MIN)):
        _warn_old_minor_version('Qt5', QT_VERSION, QT5_VERSION_MIN)
    elif QT6 and (parse(QT_VERSION) < parse(QT6_VERSION_MIN)):
        _warn_old_minor_version('Qt6', QT_VERSION, QT6_VERSION_MIN)

if PYQT_VERSION:
    if PYQT5 and (parse(PYQT_VERSION) < parse(PYQT5_VERSION_MIN)):
        _warn_old_minor_version('PyQt5', PYQT_VERSION, PYQT5_VERSION_MIN)
    elif PYQT6 and (parse(PYQT_VERSION) < parse(PYQT6_VERSION_MIN)):
        _warn_old_minor_version('PyQt6', PYQT_VERSION, PYQT6_VERSION_MIN)
elif PYSIDE_VERSION:
    if PYSIDE2 and (parse(PYSIDE_VERSION) < parse(PYSIDE2_VERSION_MIN)):
        _warn_old_minor_version('PySide2', PYSIDE_VERSION, PYSIDE2_VERSION_MIN)
    elif PYSIDE6 and (parse(PYSIDE_VERSION) < parse(PYSIDE6_VERSION_MIN)):
        _warn_old_minor_version('PySide6', PYSIDE_VERSION, PYSIDE6_VERSION_MIN)
```